src/Arcus.Scripting.Tests.Unit/Arcus.Scripting.Management.tests.ps1

```diff
@@ -0,0 +1,199 @@
+                } -Verifiable
+
+                # Act
+                { 
+                   Restore-AzApiManagementSoftDeletedService -Name $Name
+                } | Should -Throw -ExpectedMessage "API Management instance with name '$Name' is not listed as a soft deleted service and therefore it cannot be restored"
+
+                # Assert
+                Assert-VerifiableMock
+                Assert-MockCalled Get-AzCachedAccessToken -Scope It -Times 1
+            }
+            It "Restoring a soft deleted API Management should fail" {
+                # Arrange
+                $Name = 'existing-apim-instance'
+
+                Mock Get-AzCachedAccessToken -MockWith {
+                    return @{
+                        SubscriptionId = "123456"
+                        AccessToken = "accessToken"
+                    }
+                }
+
+                Mock Invoke-RestMethod {
+                    if ($Method -eq "Get") {
+                       return [pscustomobject] @{
+                        value = @([ordered] @{
+                                id = "subscriptions/########-####-####-####-############/providers/Microsoft.ApiManagement/locations/westeurope/deletedservices/$Name";
+                                name = $Name;
+                                location = "West Europe";
+                            })
+                        };
+                    } else {
+                        Throw 'some error'
+                    }
+                } -Verifiable
+
+                # Act
+                { 
+                   Restore-AzApiManagementSoftDeletedService -Name $Name
+                } | Should -Throw -ExpectedMessage "The soft deleted API Management instance '$Name' could not be restored. Details: some error"
+
+                # Assert
+                Assert-VerifiableMock
+                Assert-MockCalled Get-AzCachedAccessToken -Scope It -Times 1
+                Assert-MockCalled Invoke-RestMethod -Times 2
+            }
+            It "Restoring a soft deleted API Management should succeed" {
+                # Arrange
+                $Name = 'existing-apim-instance'
+
+                Mock Get-AzCachedAccessToken -MockWith {
+                    return @{
+                        SubscriptionId = "123456"
+                        AccessToken = "accessToken"
+                    }
+                }
+
+                Mock Invoke-RestMethod {
+                    if ($Method -eq "Get") {
+                       return [pscustomobject] @{
+                        value = @([ordered] @{
+                                id = "subscriptions/########-####-####-####-############/providers/Microsoft.ApiManagement/locations/westeurope/deletedservices/$Name";
+                                name = $Name;
+                                location = "West Europe";
+                            })
+                        };
+                    } else {
+                        return $null
```

Comment:
Is it an idea to also assert on the HTTP method here? That it should always be DELETE? And in the restore-case PUT?

Reply:
Added this to the assertion on the mock